kits/beta-base/contracts/BetaKitBase.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity 0.4.24;
+
+import "@aragon/os/contracts/factory/DAOFactory.sol";
+import "@aragon/os/contracts/kernel/Kernel.sol";
+import "@aragon/os/contracts/acl/ACL.sol";
+import "@aragon/apps-shared-minime/contracts/MiniMeToken.sol";
+
+import "@aragon/id/contracts/IFIFSResolvingRegistrar.sol";
+
+import "@aragon/apps-voting/contracts/Voting.sol";
+import "@aragon/apps-vault/contracts/Vault.sol";
+import "@aragon/apps-token-manager/contracts/TokenManager.sol";
+import "@aragon/apps-finance/contracts/Finance.sol";
+
+import "@aragon/os/contracts/common/IsContract.sol";
+
+import "@aragon/kits-bare/contracts/KitBase.sol";
+
+
+contract BetaKitBase is KitBase, IsContract {
+    MiniMeTokenFactory public minimeFac;
+    IFIFSResolvingRegistrar public aragonID;
+    bytes32[4] public appIds;
+
+    mapping (address => address) tokenCache;
+
+    // ensure alphabetic order
+    enum Apps { Finance, TokenManager, Vault, Voting }
+
+    event DeployToken(address token, address indexed cacheOwner);
+    event DeployInstance(address dao, address indexed token);
+
+    constructor(
+        DAOFactory _fac,
+        ENS _ens,
+        MiniMeTokenFactory _minimeFac,
+        IFIFSResolvingRegistrar _aragonID,
+        bytes32[4] _appIds
+    )
+        KitBase(_fac, _ens)
+        public
+    {
+        require(isContract(address(_fac.regFactory())));
+
+        minimeFac = _minimeFac;
+        aragonID = _aragonID;
+        appIds = _appIds;
+    }
+
+    function createDAO(
+        string name,
+        MiniMeToken token,
+        address[] holders,
+        uint256[] stakes,
+        uint256 _maxTokens
+    )
+        internal
+        returns (Kernel)
+    {
+        require(holders.length == stakes.length);
+
+        Kernel dao = fac.newDAO(this);
+
+        ACL acl = ACL(dao.acl());
+
+        acl.createPermission(this, dao, dao.APP_MANAGER_ROLE(), this);
```

Comment:
Can save `dao.APP_MANAGER_ROLE()` and some other roles that are used more than once as a local variable so we don't do an extra call? If it triggers `stack too deep error` we can leave it for later.